### DISCUSSION TITLE
[RFC] rootProps for each component

### DISCUSSION THREAD

**Discussion by zombieJ** [ant-design/ant-design#40227] [RFCs]
# 📋 RFC

## Summary

Each components that has validate dom support `rootClassName` `rootStyle` which always point to root dom element.

## Basic example

```tsx
// Already realized in antd
<Modal rootClassName="my-root" />
```

## Motivation

Not all of the component point `className` or `style` to root dom element.

## Detailed

To make this RFC done:
1. Add test case for preparation  
2. Add `rootClassName` `rootStyle` for each component

## Drawbacks

Some simple component not need this but for api align we add it.

## Adoption strategy

feature in minor

**Comment by zombieJ** [comment]
`rootClassName` first. `rootRef` is not well designed since some component provides multiple root dom elements. For `rootStyle` is OK but not need implement immediately.

**Comment by AM-young-fun** [comment]
root*的属性可以解决大量生产工程的兼容性问题；+1
